### ISSUE TITLE
Identity Server 4 login working in Control Panel but not in Public Site

### ISSUE THREAD

**Issue by ax2000** [verbb/social-login#26]
### Describe the bug

For context, this issue is related with this [new provider you kindly created recently.](https://github.com/verbb/social-login/issues/21)

When using ID4 login provider via Control Panel everything works fine, but it seems that the Public Site Login fails somewhere. 

When trying to login in the public site:

1. It redirects correctly to ID4 where I authenticate successfully.
2. Then it redirects me back to the public site and nothing happens.
3. If I try login again, the page just refreshes, which I think it confirms that I'm correctly logged in ID4 and get redirected back to the public site.
4. If then I try and visit the Control Panel, the login page shows and by clicking in the Social Login ID4 icon I get logged in successfully without having to authenticate in ID4,  which again I think it confirms that I've been correctly logged in ID4 after the first step.

In the logs, I can see this error:
```
2024-04-19 14:57:26 [ERROR] Unable to process callback for “auth0”: “Cannot access offset of type string on string” /var/www/html/vendor/verbb/auth/src/clients/auth0/provider/Auth0.php:81
2024-04-19 14:57:26 [INFO] Request context:
$_GET = [
    'code' => '30B71C918AE0425461E2F30C88E21C969D08EAEF5D99AA2C9835FD1C6E6ED1B6'
    'scope' => 'openid profile email'
    'state' => 'ee2e80cc4f0202466f79cb87051896dc'
    'session_state' => 'WKynwF2Jby4I9m1P3J0igIa53xLa14up7bh2Qykd9EA.D3E220E4C505F5B2EF62A8A57764EEA8'
]
```

### Steps to reproduce

As per above

### Craft CMS version

4.8.9

### Plugin version

1.0.10

### Multi-site?

No

**Comment by engram-design** [comment]
There should be very little difference between control panel and site login. I can see that it's throwing an error in the response check, which would be good to see, as there's an error coming from ID4.

It's tricky for me to debug without having access to ID4, but maybe you can add some debugging code to help see what the response is, and any error.

If you want to add the following just before [this line](https://github.com/verbb/auth/blob/98715e3635381f72e417d481c288c513bfea9c23/src/clients/auth0/provider/Auth0.php#L79)

```php
\Craft::dd($data);
```

That should output the data including any error messages from ID4.

**Comment by ax2000** [comment]
Thanks very much for looking into this Josh.

When dumping the `$data` variable before the line you requested I only get an empty string as per screenshots below:

![Screenshot 2024-04-22 at 12 16 31](https://github.com/verbb/social-login/assets/2585959/908f506f-655a-4e45-96fe-e0de8f97f580)
![Screenshot 2024-04-22 at 12 17 17](https://github.com/verbb/social-login/assets/2585959/d99877b0-4e6b-4ff0-ab67-cd984c27adfe)

I'm having a chat with the ID4 devs as well and see if it's something that they can figure out on their side.

**Comment by engram-design** [comment]
Huh, well that's unhelpful! I'll get that handling fixed up regardless as the Auth0 expects the returned value to be an array.

If you change that line to:

```php
\Craft::dd((string)$response->getBody());
```

Is there any insight there?

**Comment by ax2000** [comment]
If I change it to `\Craft::dd((string)$response->getBody());` it still shows an empty string.

If I dump the whole `$response` I get as per below screenshot:

![Screenshot 2024-04-22 at 12 52 45](https://github.com/verbb/social-login/assets/2585959/0ad4992d-6a95-46c6-94c3-cb3641c1df09)

Not sure if that's very helpful, but if I can provide anything else don't hesitate letting me know. I'm still waiting to hear back from the ID4 devs as well.

**Comment by ax2000** [comment]
Hi Josh,

Thanks for helping me with this.

After going through the whole process with the ID4 devs we spotted the issue in my template code, where I was still using the old `auth0` provider when connecting via the public website.

I could have sworn that I updated the provider handle to `identityServer4` everywhere but I missed it there.

Thanks for helping me figuring out and apologies for nit realising about this earlier.

Cheers,
Alby
